### ISSUE TITLE
Setting custom user attributes starting with uppercase

### ISSUE THREAD

**Issue by ljvanschie** [okta/okta-sdk-dotnet#159]
When creating users using `client.Users.CreateUserAsync()`, properties that start with an uppercase character will cause validation errors. The SDK serializes them to lowercase JSON fields, causing the API to return HTTP 400.

It would be nice if the casing would be maintained, or if an exception would be thrown when using attributes starting with an uppercase letter.
Furthermore, the thrown OktaApiException does not contain the detailed error cause which is available within the JSON response. It only contains the message 'Api validation failed: newUser'.

For example, in this code

```csharp
var client = GetOktaClient();
var profile = CreateUserProfile();

profile["MyAttribute"] = "value";

client.Users.CreateUserAsync(new CreateUserWithPasswordOptions
{
    Profile = profile,
    Password = "Password01",
    Activate = true
});
```

the attribute 'MyAttribute' will be transformed to 'myAttribute'.

**Comment by nbarbettini** [comment]
Thanks for the detailed bug report. I'll look into it.

**Comment by nbarbettini** [comment]
Sorry for the long delay on this @ljvanschie. This is definitely a bug and I've written a fix in #242.
